### PR TITLE
ENT-4228: Always format restlib exceptions

### PR DESCRIPTION
**Pull request by m-horky**
* Card ID: ENT-4228

Following PR #2706 (ab97ef6, ENT-3914) this change formats the rest of
rhsmlib exceptions with ExceptionMapper. It ensures that the exceptions
have unified format of "Message from candlepin. (HTTP error code XXX: YYY)".

### CODE REVIEW

File: src/subscription_manager/cli_command/register.py [hunk 1]
```diff
@@ -233,9 +235,14 @@ def _do_command(self):
                     type=self.options.consumertype,
                     service_level=self.options.service_level,
                 )
-        except (connection.RestlibException, exceptions.ServiceError) as re:
```
Comment (ptoscano) [inline src/subscription_manager/cli_command/register.py:236]: hm why not simply add `ServiceError` to the `ExceptionMapper`? this why there is no separate handling, and it is easier to change it in the future. IMHO a simple implementation that calls `str(e)` should do it

**Comment by m-horky** [comment]
I added second commit. The comment from @ptoscano lead me into a rabbit hole where, while it would be possible to just add another case to the mapping, I tried to think of more elegant solution. Let me know if you have any objections.

File: src/subscription_manager/cli_command/register.py (hunk 1 above)
Comment (m-horky) [reply src/subscription_manager/cli_command/register.py:236]: I've done some changes, see the comment and the second commit.

**Review by ptoscano — CHANGES_REQUESTED** [review]
The improvements to `ExceptionMapper`are generally nice, I only added few comments to it. The rabbit hole indeed worked, thanks :-D

The only thing I'm wondering is about the changed behaviour of `get_message()`, as it always stringifies something now: https://github.com/candlepin/subscription-manager/blob/f1b74f5c291421cc72a68267ea7abeb8321be784/src/subscription_manager/cli_command/cli.py#L66-L71
is this check still needed? could it happen that the return value is None or "false"? AFAICS, `system_exit()` should handle None as message just fine. (I'm fine about keeping it if needed, I merely wonder whether this check becomes de-facto dead code.)

**Review by ptoscano — APPROVED** [review]
Very nice, thanks!

**Pull request merged**